### PR TITLE
chore(docs): update DatePicker docs

### LINKED ISSUE #6688: Update docs for DatePicker
This issue is to update the stories that we have for the <DatePicker> component in our storybook. This work includes the following:

- [x] Update each story in `<DatePicker>-story.js` to Component Story Format
  - There is an optional codemod that you can use from storybook:
  - `npx -p @storybook/cli@next sb migrate storiesof-to-csf --glob='packages/react/src/<Component>/<Component>-story.js'`
- [ ] Update stories in the existing file, including:
  - [x] Create a default example named after the component without any knobs (should match the first example in the overview section)
  - [x] Clean-up or streamline any stories that seem to need it
  - [x] Create a `Playground` story that uses knobs for the component
    - For any other components that use knobs, add a `<DatePicker> Playground` story
- [x] Create a `<DatePicker>.mdx` file and fill it out according to the template below
  - You should import the `mdx` file in your story and pass it as an option to the story config (see example below)
  - You can refer to stories by their story id from `<DatePicker>-story.js` in `<DatePicker>.mdx`
- [x] Document any usage mishaps for v11

<details>
<summary>Template</summary>

```mdx
import { Story, Preview, Props } from '@storybook/addon-docs/blocks';

# <Component name>

## Overview
<!--
This should include the default use-case(s) for a component, at most 2.
This should be the most common way the component is used.
It should summarize what this common is for and provide snippets for how
to accomplish these common use-cases
-->

<Preview>
  <Story id="story-id" />
</Preview>

## [Use case, for example, Skeleton state or Sorting]
<!--
Each use-case of this component becomes its own section. For example,
if the component had a skeleton then you would name this section Skeleton.

If the component had a specific piece of functionality or behavior, like
how you can sort with a Data Table, then Sorting would be the name of this
section.

This section should include an explainer for the use-case and when to use it,
along with a code snippet that someone could bring into their project.
-->

## Component API

<Props />

### [Component name] [prop name]
<!--
Used to clarify prop usage, provide additional info or snippets
-->


## References
<!--
A place for references, links, or a way clarify points made earlier
in the documentation.
-->

## Feedback

Help us improve this component by providing feedback through, asking questions on Slack, or
updating this file on GitHub
[GitHub](https://github.com/carbon-design-system/carbon/edit/master/packages/react/src/components/<Component>/<Component>.mdx).
```
</details>

<details>
<summary>Configure story to use MDX page</summary>

```bash
import mdx from './Component.mdx';

export default {
  title: 'Component',
  component: Component,
  parameters: {
    docs: {
      page: mdx,
    },
  },
}
```
</details>

### PR DESCRIPTION
**Pull request by tw15egan**
Closes https://github.com/carbon-design-system/carbon/issues/6688

Updates `DatePicker` to new CSF format and adds in prop documentation 

**Done**
- [x] Update to CSF
- [x] Create `DatePicker.mdx` and fill out with template
- [x] Document any discrepancies for v11

### Changelog

**New**

### Questions for v11

- `short` prop does not seem to be needed anymore. 

#### Testing / Reviewing

Check updated docs for grammar usage and that the examples look correct

### CODE REVIEW

**Review by dakahn — APPROVED** [review]
Great stuff! :surfer:

File: packages/react/src/components/DatePicker/DatePicker.js [hunk 1]
```diff
@@ -151,58 +151,58 @@ export default class DatePicker extends Component {
...
-     * * `sq` - Albanian
-     * * `sr` - Serbian
-     * * `sv` - Swedish
-     * * `th` - Thai
-     * * `tr` - Turkish
-     * * `uk` - Ukrainian
-     * * `vn` - Vietnamese
-     * * `zh` - Mandarin
-     */
+                    *  The language locale used to format the days of the week, months, and numbers. The full list of supported locales can be found here https://github.com/flatpickr/flatpickr/tree/master/src/l10n
```
Comment (emyarod) [inline packages/react/src/components/DatePicker/DatePicker.js:154]: is all of this extra indentation intentional?

**Review by emyarod — COMMENTED** [review]
looks good to me, just wondering are we going to have content reviews like when we were rewriting the docs for the website or has it already been approved?

File: packages/react/src/components/DatePicker/DatePicker.js (hunk 1 above)
Comment (tw15egan) [reply packages/react/src/components/DatePicker/DatePicker.js:154]: Yeah, this makes them render inline in the PropTable as opposed to 1 on a line which caused the table to render really oddly.

**Comment by tw15egan** [comment]
Could be tedious to get content review on each dev docs PR, might be best to have Content do a pass over all the docs once they are up and make changes from there.

**Review by emyarod — APPROVED** [review]

**Pull request merged**